### PR TITLE
Feat  add getter and setter

### PR DESCRIPTION
**Pull request by rise-erpelding**
## Spec

See Issue: [ISSUE_NUMBER](https://sparkbox.atlassian.net/browse/FSA2021-ISSUE_NUMBER)

## Validation

- [ ] This PR has code changes, and our linters still pass.
- [ ] This PR has new code, so new tests were added or updated, and they pass.

### To Validate

1. Make sure all PR Checks have passed.
1. Pull down all related branches.
1. Navigate to... _[continue instructions here]_

### CODE REVIEW

File: src/js/Game/game.js [hunk 1]
```diff
@@ -1,12 +1,30 @@
 export default class Game {
   constructor() {
-    this.currentPlayer = 'X';
+    this._currentPlayer = 'X';
     this.winner = null;
-    this.gameBoard = new Array(9).fill(null);
+    this._gameBoard = new Array(9).fill(null);
```
Comment (jordanleven) [inline src/js/Game/game.js:5]: 😍 Using the underscore to indicate these are private properties is so awesome! Great job - my heart is singing!

File: src/js/Game/game.js [hunk 2]
```diff
@@ -1,12 +1,30 @@
...
+    return this._currentPlayer;
+  }
+
+  set currentPlayer(newPlayer) {
+    this._currentPlayer = newPlayer;
+  }
+
+  get gameBoard() {
+    return this._gameBoard;
+  }
```
Comment (jordanleven) [inline src/js/Game/game.js]: 🏆 THIS IS SO CLEAN AND CONCISE I LOVE IT! GREAT JOB!

File: src/js/Game/game.js [hunk 3]
```diff
@@ -1,12 +1,30 @@
 export default class Game {
   constructor() {
-    this.currentPlayer = 'X';
+    this._currentPlayer = 'X';
     this.winner = null;
```
Comment (jordanleven) [inline src/js/Game/game.js]: Thoughts on updating this fella?

File: src/js/Game/game.js [hunk 4]
```diff
@@ -1,12 +1,30 @@
...
+
+  set gameBoard(newBoard) {
+    this._gameBoard = newBoard;
   }
 
   markBoard(index) {
-    this.gameBoard[index] = this.currentPlayer;
+    const board = this.gameBoard;
+    board[index] = this.currentPlayer;
+    this.gameBoard = board;
```
Comment (jordanleven) [inline src/js/Game/game.js]: Out of curiousity, does somehting like `this.gameBoard[index] = this.currentPlayer` not work?

**Review by jordanleven — COMMENTED** [review]
This is really slick, @rise-erpelding! I really love the work you've done to get the getters/setters in place and especially LOVE your adoption of pretending underscores to indicate a property or method is private. Well done!

I left a few comments and a question. Let me know what you think!

File: src/js/Game/game.js (hunk 4 above)
Comment (rise-erpelding) [reply src/js/Game/game.js]: I knew there was a way to do all of that in one line, I don't know why I didn't think to try that!

Comment (jordanleven) [reply src/js/Game/game.js]: Resolved in 30811f606919e3b7c0c1f84430fb1ae7b8074b75.

File: src/js/Game/game.js (hunk 3 above)
Comment (jordanleven) [reply src/js/Game/game.js]: Resolved in 9a2bb4518d000e0e2bc04ca69a4534619f65ffcd.

**Review by jordanleven — APPROVED** [review]
This looks great! Thanks for resolving those two issues! Feel free to squash.
giphy-5

**Pull request merged**